### ISSUE TITLE
Add support for most of the common JavaScript events

### ISSUE THREAD

**Issue by Badgerati** [Badgerati/Pode.Web#156]
Most elements support JavaScript events like `onchange`, `onblur`, etc., and it'd be useful to have some of the more common ones supported in Pode.Web. These would work like a Button's ClickScriptBlock, and when the event triggers they'll call a route and get output actions.

The events I want to initially support are:
* onchange
* onfocus
* onblur
* onclick
* onload
* onmouseover
* onmouseout
* onkeydown
* onkeyup

There's ideally two ways this could be implemented:

1. We add all of these as parameters to every element and layout. They'll be scriptblocks, that create routes that can be invoked when the event triggers:
```powershell
New-PodeWebSelect -Name 'Role' -Options Dev, QA, SysAdmin, DevOps, Sales -OnChange {
    Show-PodeWebToast -Message '<message>'
}
```

2. Or, (if you're like me and the above option seems crazy 😋), we could instead add a new `Register-PodeWebEvent` function where you pipe an element/layout into it, and this binds the event instead:
```powershell
New-PodeWebSelect -Name 'Role' -Options Dev, QA, SysAdmin, DevOps, Sales |
    Register-PodeWebEvent -Type Change -ScriptBlock {
        Show-PodeWebToast -Message '<message>'
    }
```

I'm more going for the second option, to prevent mass amounts of repeating code.

**Comment by Badgerati** [comment]
Pushed some initial work for the second way - using `Register-PodeWebEvent`. Mostly hardcoded, and only for a select element.

I'm thinking about changing element/layout endpoints to be `/components/<type>/<id>` instead of `/elements/<type>/<id>` or `/layouts/<type>/<id>`. The former would make logic easier.

**Comment by Badgerati** [comment]
I've removed `onload`, and changed `onblur` to `onfocusout`.

There's been an internal refactor of property names, and URL paths.

All elements that can support events appropriately I've enabled support. The only thing really left is to possibly send the current value of the element (like a textbox value) with the event request. For example, `change` triggers on a select element, and the scriptblock is also sent the select element's current value.
